GTFOHax/hacks/aimbot.cpp

```diff
@@ -10,18 +10,232 @@ namespace Aimbot
+        std::lock_guard<std::mutex> lock(hitGhostMtx);
+        hitGhosts.emplace_back(bones, settings.hitGhostDuration);
+        
+        // Limit max number of ghosts to prevent memory issues
+        if (hitGhosts.size() > 50)
+        {
+            hitGhosts.erase(hitGhosts.begin());
+        }
+    }
+    
+    void CleanupHitGhosts()
+    {
+        std::lock_guard<std::mutex> lock(hitGhostMtx);
+        hitGhosts.erase(
+            std::remove_if(hitGhosts.begin(), hitGhosts.end(), 
+                [](const HitGhost& ghost) { return ghost.IsExpired(); }),
+            hitGhosts.end()
+        );
+    }
+    
+    void RenderHitGhosts()
+    {
+        if (!settings.hitGhostEnabled || !settings.magicBullet)
+            return;
+            
+        CleanupHitGhosts();
+        
+        std::lock_guard<std::mutex> lock(hitGhostMtx);
+        
+        for (const HitGhost& ghost : hitGhosts)
+        {
+            float alpha = ghost.GetAlpha();
+            if (alpha <= 0.0f)
+                continue;
+                
+            // Create color with fading alpha
+            ImVec4 color = settings.hitGhostColor;
+            color.w *= alpha;
+            ImU32 colorU32 = ImGui::GetColorU32(color);
+            
+            // Draw skeleton bones for each group
+            for (const auto& boneGroup : Enemy::skeletonBones)
+            {
+                app::Vector3 prevPos = {0, 0, 0};
+                bool hasPrev = false;
+                
+                for (const auto& boneType : boneGroup)
+                {
+                    auto it = ghost.skeletonPositions.find(boneType);
+                    if (it == ghost.skeletonPositions.end())
+                        continue;
+                        
+                    app::Vector3 curPos = it->second;
+                    
+                    if (hasPrev && !(prevPos.x == 0.0f && prevPos.y == 0.0f && prevPos.z == 0.0f))
+                    {
+                        // Draw bone line
+                        ImVec2 screenStart, screenEnd;
+                        if (Math::WorldToScreen(prevPos, screenStart) && 
+                            Math::WorldToScreen(curPos, screenEnd))
+                        {
+                            ImGui::GetBackgroundDrawList()->AddLine(
+                                screenStart, screenEnd, colorU32, settings.hitGhostThickness);
+                        }
+                    }
+                    
+                    prevPos = curPos;
+                    hasPrev = true;
+                }
+            }
+        }
+    }
+    
+    void AddBulletRay(app::Vector3 startPos, app::Vector3 endPos)
+    {
+        if (!settings.bulletRayEnabled)
+            return;
+            
+        std::lock_guard<std::mutex> lock(bulletRayMtx);
+        bulletRays.emplace_back(startPos, endPos, settings.bulletRayDuration);
+        
+        // Limit max number of rays to prevent memory issues
+        if (bulletRays.size() > 50)
```

Comment:
Have you tested this? I don't see how 50+ rays would cause a memory issue. Maybe it could look overwhelming but I don't think so.

Reply:
This section originally used suboptimal logic during initial development. I later addressed these issues, so the comments were simply forgotten to be removed. There are currently no performance concerns here, though excessive rays can indeed cause clutter. That's why this restriction hasn't been removed.